kitsune/users/views.py

```diff
@@ -759,3 +773,107 @@ def get_settings(attr, *args):
         if val is not None:
             return val
         return super(FXALogoutView, FXALogoutView).get_settings(attr, *args)
+
+
+class WebhookView(View):
+    """The flow here is based in the mozilla-django-oidc lib.
+
+    If/When the said lib supports SET tokens this will be replaced by the lib.
+    """
+    @staticmethod
+    def get_settings(attr, *args):
+        """Override the logout url for Firefox Accounts."""
+
+        val = get_oidc_fxa_setting(attr)
+        if val is not None:
+            return val
+        return import_from_settings(attr, *args)
+
+    def verify_token(self, token, **kwargs):
+        """Validate the token signature."""
+
+        token = force_bytes(token)
+        key = self.OIDC_RP_CLIENT_SECRET
+
+        jws = JWS.from_compact(token)
+
+        try:
+            alg = jws.signature.combined.alg.name
+        except KeyError:
+            msg = 'No alg value found in header'
+            raise SuspiciousOperation(msg)
+
+        if alg != self.OIDC_RP_SIGN_ALGO:
+            msg = "The provider algorithm {!r} does not match the client's " \
+                  "OIDC_RP_SIGN_ALGO.".format(alg)
+            raise SuspiciousOperation(msg)
+
+        if isinstance(key, six.string_types):
+            # Use smart_bytes here since the key string comes from settings.
+            jwk = JWK.load(smart_bytes(key))
+        else:
+            # The key is a json returned from the IDP JWKS endpoint.
+            jwk = JWK.from_json(key)
+
+        if not jws.verify(jwk):
+            msg = 'JWS token verification failed.'
+            raise SuspiciousOperation(msg)
+
+        # The 'token' will always be a byte string since it's
+        # the result of base64.urlsafe_b64decode().
+        # The payload is always the result of base64.urlsafe_b64decode().
+        # In Python 3 and 2, that's always a byte string.
+        # In Python3.6, the json.loads() function can accept a byte string
+        # as it will automagically decode it to a unicode string before
+        # deserializing https://bugs.python.org/issue17909
+        return json.loads(jws.payload.decode('utf-8'))
+
+    def post(self, request, *args, **kwargs):
+        # TODO add docstrings
+        # TODO add error handling
+
+        error = ''
+        description = ''
+        id_token = kwargs.get('id_token')
+        # SET event must be of type `application/secevent+jwt`
+        if request.META.get('CONTENT_TYPE', '') != 'application/secevent+jwt':
+            raise Http404
+
+        payload = self.verify_token(id_token)
+
+        if payload:
+            issuer = payload['iss']
+            event = payload.get('events', '')
+            fxa_uid = payload.get('sub', '')
+            exp = payload.get('exp')
```

Comment:
Please add some docs or check keys instead. It's not really intuitive what's the security consideration here.